components/mintKamon/GenerateImageForm.tsx

```diff
@@ -1,13 +1,96 @@
-import { Button, Heading } from "@chakra-ui/react"
-import { Layout } from "@/components/layouts/layout"
+import { useAccount } from 'wagmi'
+import { useState } from 'react'
+import { Button, Heading, Text, FormControl, FormLabel, Input } from '@chakra-ui/react'
+import { Layout } from '@/components/layouts/layout'
 
 const GenerateImageForm = () => {
+  const ipfsApiEndpoint = 'https://api.staging.sakazuki.xyz/henkaku/ipfs'
+  const { data } = useAccount()
+  const [input, setInput] = useState('')
+  const [disabled, setDisabled] = useState(false);
+  const handleInputChange = (e: any) => setInput(e.target.value)
+  const handleFileChange = (e: any) => setInput(e.target.files[0])
+  const isError = input === ''
+
+  const submitGenerateImage = async (event: any) => {
+    event.preventDefault()
+    setDisabled(true)
+    const imageFile = event.target.profilePicture.files[0]
+    const blobToBase64 = (blob: any): Promise<string> => {
+      return new Promise<string>((resolve, _) => {
+        const reader = new FileReader()
+        reader.onloadend = () => {
+          if (typeof reader.result == 'string') {
+            resolve(reader.result)
+          } else {
+            resolve('')
+          }
+        }
+        reader.readAsDataURL(blob)
+      })
+    }
+    const payLoad = {
+      name: event.target.name.value,
+      address: data?.address,
+      roles: ['Community member'],
+      points: 100,
+      date: ~~(new Date().getTime() / 1000),
+      profilePicture: await blobToBase64(imageFile),
+    }
+
+    const res = await fetch(
+      ipfsApiEndpoint,
+      {
+        body: JSON.stringify(payLoad),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        method: 'POST',
+      },
+    )
+    const result = await res.json()
+    const tokenURI = result.tokenUri
+    console.log(tokenURI)
+  }
```

Comment:
I think we can use like useHooks or other to refactor this component.
what do you think ? @hourglasshoro 
is there better way to accomplish this component ?

Reply:
It is true that refactoring might be cleaner, but the front can be changed at any time, so if you are in a hurry, you might as well leave it as it is for the time being. I think it would be good if it is commonized when it is used in other places again.

Reply:
yeah true !